epitome/test/generators_test.py

```diff
@@ -104,165 +104,170 @@ def test_generator_only_H3(self):
+		features = results[0][0]
+
+		# get labels
+		labels = results[0][1]
+
+		#  all cell types but K562 are missing LARP7 data
+		assert(len(features[0]) == len(eligible_cells) * len(eligible_targets) - 3)
+
+		# make sure mask is masking out LARP7 for HepG2
+		assert(np.all(features[-1] == [1., 0., 1.]))
+
+		# make sure first label cell is not the test cell K562
+		assert(labels[-2][0] == 'lbl_HepG2_RAD21')
+		assert(labels[-2][1] == 'lbl_HepG2_LARP7')
+		assert(labels[-2][2] == 'lbl_HepG2_CTCF')
+
+	def test_generator_radius(self):
+		eligible_cells = ['K562','HepG2','H1','A549','HeLa-S3']
+		eligible_targets = ['DNase','CTCF','RAD21']
+
+		dataset = EpitomeDataset(targets = eligible_targets,
+			cells = eligible_cells,
+			data_dir=self.epitome_test_path)
+
+		label_cell_types = ['K562']
+		eligible_cells.remove(label_cell_types[0])
+
+		radii = [1,10]
+
+		results = load_data(dataset.get_data(Dataset.TRAIN),
+			['K562'],
+			eligible_cells,
+			dataset.matrix,
+			dataset.targetmap,
+			dataset.cellmap,
+			radii = radii,
+			mode = Dataset.VALID,
+			indices=np.arange(0,10))()
+		li_results = list(results)
+
+		# length should include eligible targets and 2* radius for pos and agreement
+		assert(len(li_results[0][0]) == len(eligible_cells) * (len(eligible_targets)+len(radii)* 2))
+
+	def test_generator_multiple_sim(self):
+		eligible_cells = ['K562','HepG2','H1','A549','HeLa-S3']
+		eligible_targets = ['DNase','CTCF','RAD21']
+		dataset = EpitomeDataset(targets = eligible_targets,
+					cells = eligible_cells,
+					data_dir=self.epitome_test_path)
+
+		label_cell_types = ['K562']
+		eligible_cells.remove(label_cell_types[0])
+
+		similarity_matrix = np.ones([2,dataset.get_data(Dataset.TRAIN).shape[1]])
+
+		radii = [1,10]
+
+		results = load_data(dataset.get_data(Dataset.TRAIN),
+			['K562'],
+			eligible_cells,
+			dataset.matrix,
+			dataset.targetmap,
+			dataset.cellmap,
+			radii = radii,
+			mode = Dataset.RUNTIME,
+			similarity_matrix = similarity_matrix,
+			similarity_targets = ['DNase','CTCF'],
+			indices=np.arange(0,10))()
+		li_results = list(results)
+
+		# length should include eligible targets and 2* radius for pos and agreement
+		# for each of the 2 similarity targets
+		assert(len(li_results[0][0]) == len(eligible_cells) * (len(eligible_targets)+len(radii)* 4))
+
+	def test_generator_dnase_array(self):
+		# should not fail if similarity_targets are just for DNase and is a single array.
+		# https://github.com/YosefLab/epitome/issues/4
+		eligible_cells = ['K562','HepG2','H1','A549','HeLa-S3']
+		eligible_targets = ['DNase','CTCF','RAD21','LARP7']
+		dataset = EpitomeDataset(targets = eligible_targets,
+							cells = eligible_cells,
+							data_dir=self.epitome_test_path)
+		test_celltypes = ['K562']
+		eligible_cells.remove(test_celltypes[0])
+		radii = [1,10]
+		# fake data for DNase
+		similarity_matrix = np.ones(dataset.get_data(Dataset.TRAIN).shape[1])
+
+		results = load_data(dataset.get_data(Dataset.TRAIN),
```

Comment:
what is going on here?

Reply:
I'm not too sure, I didn't alter these tests except I added the `data_dir` path for all the new Dataset cases